lib/srv/desktop/tdp/proto.go

```diff
@@ -1236,6 +1233,11 @@ func decodeSharedDirectoryReadResponse(in io.Reader) (SharedDirectoryReadRespons
 		return SharedDirectoryReadResponse{}, trace.Wrap(err)
 	}
 
+	if readDataLength > maxLen {
+		io.CopyN(io.Discard, in, int64(readDataLength))
```

Comment:
Explicitly intentional: 
```suggestion
		_, _ = io.CopyN(io.Discard, in, int64(readDataLength))
```